### PR TITLE
Rewrite OpenXR hand tracking to reflect new architecture

### PR DESCRIPTION
**Pull request by BastiaanOlij**
Work in progress, I'm rewriting the hand tracking manual page to reflect the new way things are structured in Godot 4.3.
About half way done.

Note that this manual page references https://github.com/godotengine/godot-demo-projects/pull/973 and that PR needs to be merged alongside this.

### CODE REVIEW

File: tutorials/xr/openxr_hand_tracking.rst
Comment (BastiaanOlij) [inline tutorials/xr/openxr_hand_tracking.rst:88]: Reminder to self, need to do this.

File: tutorials/xr/openxr_body_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_body_tracking.rst:4]:
```suggestion
====================
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:17]:
```suggestion
You may find yourself in a place where chosen hardware doesn't support a piece of the puzzle or does things differently
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:222]:
```suggestion
is a new core extension which supports pinch, grasp, and poke gestures and related poses.
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:193]:
```suggestion
If there is a tracker but `has_tracking_data` is false, the user's hand is currently not being tracked.
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:32]:
```suggestion
More recently optical hand tracking has become a popular solution, where cameras track the user's hands
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:317]:
```suggestion
user's palm is facing away from the user.
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:319]:
```suggestion
are touching while the user's palm is facing towards the user.
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:99]:
```suggestion
The hand tracking system uses separate hand trackers to track the position of the player's hands
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:27]:
```suggestion
From the tracking data we can infer the location of the player's hands but no further information is known,
traditionally it was left up to the game to implement a mechanism to display the player's hand and animate
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (AThousandShips) [inline tutorials/xr/openxr_hand_tracking.rst:109]:
```suggestion
    avatar creation UIs, fake mirrors, or similar situations
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (dsnopek) [inline tutorials/xr/openxr_hand_tracking.rst:33]:
```suggestion
and full tracking data for the hand and finger positions becomes available.
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (dsnopek) [inline tutorials/xr/openxr_hand_tracking.rst:35]:
```suggestion
access hand and finger positions and orientation data.
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (dsnopek) [inline tutorials/xr/openxr_hand_tracking.rst:36]:
```suggestion
When handling input, it was up to the game developer to implement a gesture detection mechanism.
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (dsnopek) [inline tutorials/xr/openxr_hand_tracking.rst:61]:
```suggestion
The information presented on this page was used to create a demo project that can be found
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (dsnopek) [inline tutorials/xr/openxr_hand_tracking.rst:75]: Probably not important, but I think we try to use US (vs UK) spellings:

```suggestion
The hand tracking implementation in Godot has been standardized around the Godot Humanoid Skeleton
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (dsnopek) [inline tutorials/xr/openxr_hand_tracking.rst:110]: Another US vs UK spelling:

```suggestion
    resulting in the hand mesh and finger tracking being localized elsewhere.
```

File: tutorials/xr/openxr_hand_tracking.rst
Comment (dsnopek) [inline tutorials/xr/openxr_hand_tracking.rst:213]: US vs UK spelling:

```suggestion
Recognizing that there is an increasing demand for applications that can switch seamlessly
```

**Review by dsnopek — COMMENTED** [review]
Thanks! This looks great!

I only have a couple language nit-picks.

**Comment by BastiaanOlij** [comment]
I think I got all the reported grammar issue, thanks for the proof reading @AThousandShips and @dsnopek

**Review by dsnopek — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Review by skyace65 — CHANGES_REQUESTED** [review]

**Review by skyace65 — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Review by m4gr3d — APPROVED** [review]

**Comment by BastiaanOlij** [comment]
Thanks for the approvals everyone, please also check out the demo project that accompanies this.

**Comment by skyace65** [comment]
I can check that the demo runs on my computer later this week. Unfortunately though my headset doesn't have hand tracking.

**Comment by BastiaanOlij** [comment]
> I can check that the demo runs on my computer later this week. Unfortunately though my headset doesn't have hand tracking.

If you have steamVR it'll work on inferred hand tracking, all should still work :)

**Comment by mhilbrunner** [comment]
This PR is good to go and just waiting on the demo to be merged. Unfortunately, I currently lack the bandwith to check it out myself. Please poke me once that PR is merged so we can merge this one :)

**Comment by skyace65** [comment]
XR Demo is merged as well. Thanks for your work on both of these!

**Pull request merged**
